File: Bubble/Models/Usuario.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Bubble.Models;

public class Usuario
{
    public int Id { get; set; }
    public string Nombre { get; set; }
    public string Email { get; set; }
    public string Contrasena { get; set; }
    public string Rol { get; set; }
    public DateTime FechaCreacion { get; set; } = DateTime.Now;
    public bool Activo { get; set; } = true; // Propiedad opcional
}
public class LoginRequest
{
    [Required]
    [EmailAddress]
    public string Email { get; set; } = null!;

    [Required]
    public string Password { get; set; } = null!;
}

public class RegisterRequest
{
    [Required]
    [EmailAddress]
    public string Email { get; set; } = null!;

    [Required]
    [MinLength(6)]
    public string Password { get; set; } = null!;

    [Required]
    public string Nombre { get; set; } = null!;

    public string Rol { get; set; } = "Cliente";
}

public class LoginResponse
{
    public string Token { get; set; } = null!;
    public string Email { get; set; } = null!;
    public string Nombre { get; set; } = null!;
    public string Rol { get; set; } = null!;
    public DateTime Expiracion { get; set; }
}

File: lib/models/usuario.dart
class Usuario {
  int id;
  String nombre;
  String email;
  String contrasena;
  String rol;
  DateTime fechaCreacion;
  bool activo;

  Usuario({
    this.id = 0,
    required this.nombre,
    required this.email,
    required this.contrasena,
    this.rol = 'Cliente',
    DateTime? fechaCreacion,
    this.activo = true,
  }) : fechaCreacion = fechaCreacion ?? DateTime.now();

  factory Usuario.fromJson(Map<String, dynamic> json) => Usuario(
        id: json['id'] ?? 0,
        nombre: json['nombre'] ?? '',
        email: json['email'] ?? '',
        contrasena: json['contrasena'] ?? '',
        rol: json['rol'] ?? 'Cliente',
        fechaCreacion: json['fechaCreacion'] != null
            ? DateTime.parse(json['fechaCreacion'])
            : DateTime.now(),
        activo: json['activo'] ?? true,
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'nombre': nombre,
        'email': email,
        'contrasena': contrasena,
        'rol': rol,
        'fechaCreacion': fechaCreacion.toIso8601String(),
        'activo': activo,
      };
}